File: ts-sdk/src/sdk/models/shared/signupuserpayload.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { UserDetail } from "./userdetail";


export enum SignupUserPayloadLanguageEnum {
    En = "en",
    De = "de"
}


export class SignupUserPayload extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=language" })
  language?: SignupUserPayloadLanguageEnum;

  @SpeakeasyMetadata({ data: "json, name=organization_detail" })
  organizationDetail?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=user_detail" })
  userDetail?: UserDetail;
}

File: user-api/python-sdk/src/epilot/models/shared/signupuserpayload.py
import dataclasses
from typing import Any,Optional
from enum import Enum
from dataclasses_json import dataclass_json
from epilot import utils
from ..shared import userdetail as shared_userdetail

class SignupUserPayloadLanguageEnum(str, Enum):
    EN = "en"
    DE = "de"


@dataclass_json
@dataclasses.dataclass
class SignupUserPayload:
    language: Optional[SignupUserPayloadLanguageEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('language') }})
    organization_detail: Optional[dict[str, Any]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('organization_detail') }})
    user_detail: Optional[shared_userdetail.UserDetail] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('user_detail') }})
    
